### PR TITLE
Send builder status to GitHub

### PR DESCRIPTION
**Pull request by adiroiban**
Scope
=====

This add support for reporting build result status for all builds done by Buildbot  for fixing #147

Changes
=======

I have configured GitHub status reporting in the buildbot configuration.

I have created a dedicated OAuth app in GitHub for Buildbot status
 https://github.com/organizations/twisted/settings/applications/373261

I have set its logo to the Buildbot nut logo :) Feel free to suggest any other image

For testing, I got my own token... and before deploying in production I will update the private repo with the token generated for the dedicated OAuth app

How to test
=========

You can see it in action here

* For a PR  https://github.com/twisted/twisted/pull/212
* For a branch comparison https://github.com/twisted/twisted/compare/8494-codecov-config

Once this hit production we will see 28 build results from buildbot

### CODE REVIEW

**Comment by adiroiban** [comment]
Check that start and end build messages are ok.

To actually test it we need to deploy it in production.

I can deploy it from this branch without merging it... and if there are problems I can re-deploy from master.

Please check that changes are ok. Thanks!

File: services/buildbot/master/private.py.sample [hunk 1]
```diff
@@ -65,7 +65,7 @@ codecov_twisted_token = 'invalid'
 #
 #
 # For production get a OAuth Token.
-# GitHub OAuth Application for Builbot Status is here:
+# GitHub OAuth Application for Twisted Builbot Status is here:
```
Comment (rodrigc) [inline services/buildbot/master/private.py.sample]: Since you are fixing things, might as well fix the spelling for Buildbot

**Comment by adiroiban** [comment]
I have deployed this branch in production... by manually update the private data... since I don't yet have access to the private repo.

You can see it in action here https://github.com/twisted/twisted/pull/212 ... 31 status reports in total :)

**Comment by rodrigc** [comment]
I have observed successful buildbot --> GitHub integration in my PR's such as this one:
https://github.com/twisted/twisted/pull/214

Thanks!

**Comment by adiroiban** [comment]
I have merged this I think that is working OK.

The unwritten review rule used to merge this is that if a PR is not reviewed in 1 week it can be merged :)

**Pull request merged**